File: hto/57/hto_57.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (!set.contains(target - num))
                set.add(num);
            else 
                return new int[]{num, target - num};
        }
        return new int[]{};
    }
}

File: hto/57/hto_57.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;
        while (left < right) {
            int sum = nums[left] + nums[right];
            if (sum == target)     return {nums[left], nums[right]};
            else if (sum < target) left++;
            else if (sum > target) right--;
        }
        return {};
    }
};